packages/plugin/tests/require-import-fragment.spec.ts

```diff
@@ -0,0 +1,74 @@
+import { GraphQLRuleTester } from '../src';
+import { rule } from '../src/rules/require-import-fragment';
+
+const ruleTester = new GraphQLRuleTester();
+
+ruleTester.runGraphQLTests('require-import-fragment', rule, {
+  valid: [
+    {
+      name: 'should not report imported fragments',
+      code: /* GraphQL */ `
+        # import Foo from 'foo.graphql'
+
+        query MyQuery {
+          fooField {
+            ...Foo
+          }
+        }
+      `,
+    },
+    {
+      name: 'should not report fragments from the same file',
+      code: /* GraphQL */ `
+        query MyQuery {
+          fooField {
+            ...Foo
+          }
+        }
+
+        fragment Foo on Bar {
+          baz
+        }
+      `,
+    },
+  ],
+  invalid: [
+    {
+      name: 'should report fragments when there are no import expressions',
+      code: /* GraphQL */ `
+        query MyQuery {
+          fooField {
+            ...Foo
+          }
+        }
+      `,
+      errors: [{ message: "Expected 'Foo' fragment to be imported." }],
+    },
+    {
+      name: 'should report fragments when there are no named import expressions',
+      code: /* GraphQL */ `
+        # import 'foo.graphql'
+
+        query MyQuery {
+          fooField {
+            ...Foo
+          }
+        }
+      `,
+      errors: [{ message: "Expected 'Foo' fragment to be imported." }],
+    },
```

Comment:
I guess it's a valid case? and the fragment will be still imported?

Reply:
I thought that I have seen this syntax supported somewhere, but I can't find any reference to it. So I'll adjust the test description to `should report fragments when there are only invalid import expressions`

Reply:
it's a valid import https://github.com/ardatan/graphql-tools/blob/37c71441751801e6a7bfca6966ffe7f83e7992eb/packages/import/tests/documents/import-documents.spec.ts#L9-L29

and fixture

https://github.com/ardatan/graphql-tools/blob/37c71441751801e6a7bfca6966ffe7f83e7992eb/packages/import/tests/documents/import-test/default/a.graphql#L1-L7

Reply:
Ah, I see. I guess these are our options:

1. Keep current behavior, i.e. treat default imports `# import 'foo.graphql'` as non-existent and require specific imports `# import Foo from 'foo.graphql'`.
  This would allow this rule to be enabled without information about other operation files.
  Importing `Foo` from `not-foo.graphql` would not be treated as an error by this rule, but instead be reported by a separate rule (that requires operation file information): #1416
2. Treat all fragment spreads as valid as soon as there is a default import `# import 'foo.graphql'`. This would work without operation file information, too. But which rule would then check whether `Foo` is really in `foo.graphql`?
  Maybe #1414? But then that one would need operation file information, while it currently doesn't.
3. Require operation file information in this rule and explicitly check whether default imports contain the specified fragment.
4. Add an option to this rule that specifies how to treat default imports:
  * `defaultImports: 'ignore'` = option 1
  * `defaultImports: 'check' = option 3, would then require operation file information
  * `defaultImports: 'forbid' → disallow default imports altogether. But that should probably be a separate rule.

Or some other option or a combination of the above.

Reply:
> ignore default imports # import 'foo.graphql' and require specific imports # import Foo from 'foo.graphql'.

no, we should validate the both

> This would allow this rule to be enabled without information about other operation files.

this rule should require operations files to be smart

> Add an option to this rule that specifies how to treat default imports:

I guess we don't need options


I will help you today-tomorrow, and will push directly in your PR some logic, so you could reuse for other rules

Reply:
@FloEdelmann check what I made ;)